### ISSUE TITLE
[v2] can't return XML response with content-type set to application/xml

### ISSUE THREAD

**Issue by brandonh-msft** [Azure/azure-functions-host#2896]
I was trying, in a v2 function, to return an XML response to the caller. Of course, I want my `Content-Type` header to be properly set to `application/xml` in this case.

Using the V2 (ASP.Net Core) types, I was unable to do so. Documented my experience [here](http://netitude.bc3tech.net/2018/05/25/creating-an-xml-json-converter-in-azure-functions)

#### Investigative information

- Timestamp: n/a
- Function App version (1.0 or 2.0-beta): 2.0-beta

#### Repro steps

1. Create a .Net v2 HttpTrigger Azure Function in Visual Studio.
1. Attempt to return XML payload from the response **with the `Content-Type` header properly set to `application/json`**

#### Expected behavior

I am able to return XML content easily, such as `return new OkObjectResult(xmlDoc);` or even `return new OkObjectResult(xmlDoc) { ContentTypes = new Microsoft.AspNetCore.Mvc.Formatters.MediaTypeCollection { @"application/xml" } };`

#### Actual behavior

Output is either heavily escaped, wrapped in a .Net type tag (eg: `<string>`), doesn't have the `Content-Type` header set, or throws an `HTTP 406 NOT ACCEPTABLE` error.

#### Known workarounds

Use the `System.Net.Http` types from v1 (eg: `HttpResponseMessage`) to create the response payload instead of the .Net Core (v2) types.

#### Related information 
http://netitude.bc3tech.net/2018/05/25/creating-an-xml-json-converter-in-azure-functions

**Comment by paulbatum** [comment]
cc @fabiocav 

@brandonh-msft is this a typo?

> Attempt to return XML payload from the response with the Content-Type header properly set to **application/json**

**Comment by brandonh-msft** [comment]
Thanks for the tip on `Accept` and `ContentResult`

Is there any way we could make Functions respect `ProducesAttribute` to alleviate some of this pain? I was *very* surprised when I told `OkObjectResult` that the MediaType should be `application/xml` and got back `<string> [my stuff here] </string>`

**Comment by brandonh-msft** [comment]
Gave `Accept` a shot but unfortunately, in my example of a converter, it didn't work:

~~~
POST /api/ConvertToXml HTTP/1.1
Host: localhost:7071
Content-Type: application/json
Accept: application/xml
Cache-Control: no-cache
Postman-Token: 6533fbab-15bc-49ee-881e-8249f23b364e

{
    "root": {
        "this": {
            "@att": "x",
            "underthis": "val"
        },
        "that": {
            "withval": "x",
            "bigval": {
                "#cdata-section": "\n\t\t\tsomething something\n\t\t\t"
            }
        }
    }
}
~~~
Got back
~~~
{
    "root": {
        "this": {
            "@att": "x",
            "underthis": "val"
        },
        "that": {
            "withval": "x",
            "bigval": {
                "#cdata-section": "\n\t\t\tsomething something\n\t\t\t"
            }
        }
    }
}
~~~

despite code:
```csharp
if (req.ContentType.IndexOf(@"/json", 0, System.StringComparison.OrdinalIgnoreCase) == -1)
{
    return new BadRequestObjectResult(@"Content-Type header must be a JSON content type");
}

var json = await req.ReadAsStringAsync();
XmlDocument doc = JsonConvert.DeserializeXmlNode(json);

return new OkObjectResult(doc);
```

`ContentResult`, however, did exactly what I needed.

**Comment by brandonh-msft** [comment]
awesome thanks for the xpost to SO

**Comment by brandonh-msft** [comment]
yeah I had a feeling there might be some `Startup()`-like MVC stuff that handles it in .Net Core land, but alas we don't get to see that in Functions world.
